R/github.R

```diff
@@ -3,23 +3,51 @@ isGitHubURL <- function(url) {
 }
 
 canUseGitHubDownloader <- function() {
-  all(packageVersionInstalled(devtools = "1.9.1", httr = "1.0.0"))
+  all(packageVersionInstalled(httr = "1.0.0")) & (!is.null(github_pat()))
+}
+
+github_pat <- function(quiet = TRUE) {
+  pat <- Sys.getenv("GITHUB_PAT")
+  if (nzchar(pat)) {
+    if (!quiet) {
+      message("Using GitHub username from envvar GITHUB_PAT")
+    }
+    return(pat)
+  }
+  return(NULL)
 }
 
 githubDownload <- function(url, destfile, ...) {
-  onError(1, {
-    github_pat      <- yoink("devtools", "github_pat")
-    authenticate    <- yoink("httr", "authenticate")
-    GET             <- yoink("httr", "GET")
-    content         <- yoink("httr", "content")
+  tryCatch(
+    githubDownloadImpl(url, destfile, ...),
+    error = function(e) {
+      stop(sprintf("GitHub request failed with %s", e), call. = FALSE)
```

Comment:
```suggestion
      stop(sprintf("GitHub request failed with %s", conditionMessage(e)), call. = FALSE)
```

Reply:
As above: message tweaked, but not using `conditionMessage`.